### DISCUSSION TITLE
Virtual actors and auto placement

### DISCUSSION THREAD

**Discussion by cscetbon** [tqwewe/kameo#78] [Ideas]
Thanks for this project, I'm using Dapr and I'd say some of the things that retain me there are :
- I don't need to use any address cause there is a placement component that is in charge of directing the request to the right location based on the actor ID
- I don't need to shutdown actors cause they go offline automatically after some time
- I don't need to rebalance actors cause it's done automatically

just my 2 cents

**Comment by tqwewe** [comment]
Hi thanks for sharing. I don't know that I can agree with the points you mentioned as being features over kameo. Dapr seems to have a different scope to kameo.
- Since kameo is designed for handling dozens, hundreds, or even thousands of actors within a single application, I cannot see a way around actor refs as a way to message the right ones.
- I'm not sure what cases you might want an actor to shutdown after some time specifically, but by default I'd imagine most cases would want an actor to run continuously until explicitly stopped, or if no other parts of your app need to talk to it anymore.

I don't have much insight into dapr however.
